directus-solidstart/index.md

```diff
@@ -0,0 +1,314 @@
+---
+title: "Getting Started with Directus and SolidStart"
+description: "Let's learn how to build a blog using Directus as a CMS. You will store, retrieve, and use global metadata such as the site title, create new pages dynamically based on Directus items"
+author:
+  name: "Atila Fassina"
+  avatar_file_name: "author.webp"
+---
+
+## Summary
+
+[SolidStart](https://start.solidjs.com) a meta-framework built around [SolidJS](https://solidjs.com) - SolidStart is a framework aiming to allow people and teams to built high-end applications while bringing their own opinions. In this tutorial, you will learn how to build a website using Directus as a CMS. You will store, retrieve, and use global metadata such as the site title, create new pages dynamically based on Directus items, and build a blog.
+
+## Before You Start
+
+You will need:
+
+- To install Node.js and a code editor on your computer.
+- A Directus project - follow our quickstart guide if you don't already have one.
+- SolidJS knowledge will definitely help, but will not prevent you from following this guide.
+
+## Initialize SolidStart Project
+
+Open your terminal and run the following commands to create a new Solid project and add the Directus JavaScript SDK:
+
+```sh
+npm create solid@latest solid-directus
+```
+
+A prompt will start, here are the sequence of answers provided for this exercise:
+
+```text
+✔ Which template do you want to use?
+  › bare
+
+✔ Server Side Rendering?
+  › yes
+
+✔ Use TypeScript?
+  › no
+```
+
+Now change directory to what you created and install all dependencies while adding the Directus SDK.
+
+```sh
+cd solid-directus && npm install @directus/sdk
+```
+
+With this, we are ready to start coding!
+
+```sh
+npm run dev --open
+```
+
+It will open a browser window or tab on [localhost:3000](https://localhost:3000) or the next available port.
+
+## Create a Directus Helper
+
+We now need to setup the Directus SDK. We will add our data fetchers in this same file so we can handle caching and requests all at the same place. Create a new file `directus.js` inside of the src/libs directory:
+
+```jsx
+import { createDirectus, readItem, readItems, rest } from "@directus/sdk";
+import { cache } from "@solidjs/router";
+
+export const directus = createDirectus("< your project URL >").with(rest());
+```
+
+We have some unused `import` statements at the moment since now we are only instantiating the Directus client.
+
+> Remember to replace `"< your project URL >"` with your actual URL, you can find it in your Directus dashboard.
+
+It may be useful to create an `.env` file and store the URL there in case you use different projects for production, development, etc. Otherwise it is not a private value, access is managed through the Directus dashboard.
```

Comment:
Let's remove this and assume everything is public for this post. We can mention this in __Next Steps__.